lib/IRGen/GenStruct.cpp

```diff
@@ -290,7 +291,46 @@ namespace {
       }
     }
   };
-  
+
+  class ClangRecordAddressOnlyTypeInfo final
+      : public StructTypeInfoBase<ClangRecordAddressOnlyTypeInfo, FixedTypeInfo,
+                                  ClangFieldInfo> {
+    const clang::RecordDecl *ClangDecl;
+
+  public:
+    ClangRecordAddressOnlyTypeInfo(ArrayRef<ClangFieldInfo> fields,
+                                   llvm::Type *storageType, Size size,
+                                   SpareBitVector &&spareBits, Alignment align,
+                                   const clang::RecordDecl *clangDecl)
+        : StructTypeInfoBase(StructTypeInfoKind::ClangRecordAddressOnlyTypeInfo,
+                             fields, storageType, size, std::move(spareBits),
+                             align, IsPOD, IsNotBitwiseTakable, IsFixedSize),
+          ClangDecl(clangDecl) {
+      (void)ClangDecl;
```

Comment:
I'm sure we'll need this later but, I also don't want compiler warnings.

Reply:
SGTM.